packages/drawer/src/ux-drawer.ts

```diff
@@ -0,0 +1,302 @@
+  @bindable public keyboard: DrawerKeybord = ['Escape'];
+  @bindable public restoreFocus?: (lastActiveElement: HTMLElement) => void= (lastActiveElement: HTMLElement) => {
+    lastActiveElement.focus();
+  }
+
+  // Aria attributes
+  @bindable public role: 'dialog' | 'alertdialog' = 'dialog';
+  @bindable public ariaLabelledby: string = '';
+  @bindable public ariaDescribedby: string = '';
+
+  public lastActiveElement?: HTMLElement;
+
+  private handlingEvent: boolean = false;
+  private viewportType: 'mobile' | 'desktop' = 'desktop';
+  private overlayElement: HTMLElement;
+  private contentWrapperElement: HTMLElement;
+  private contentElement: HTMLElement;
+  private showed: boolean = false;
+  private showing: boolean = false;
+  private hiding: boolean = false;
+
+  constructor(
+    public element: HTMLElement,
+    private styleEngine: StyleEngine,
+    private modalService: ModalService,
+    private taskQueue: TaskQueue,
+    private defaultConfig: DefaultDrawerConfiguration) {
+      if (this.defaultConfig.modalBreakpoint !== undefined) {
+        this.modalBreakpoint = this.defaultConfig.modalBreakpoint
+      }
+      if (this.defaultConfig.host !== undefined) {
+        this.host = this.defaultConfig.host
+      }
+      if (this.defaultConfig.overlayDismiss !== undefined) {
+        this.overlayDismiss = this.defaultConfig.overlayDismiss
+      }
+      if (this.defaultConfig.position !== undefined) {
+        this.position = this.defaultConfig.position
+      }
+      if (this.defaultConfig.keyboard !== undefined) {
+        this.keyboard = this.defaultConfig.keyboard
+      }
+      if (this.defaultConfig.theme !== undefined) {
+        this.theme = this.defaultConfig.theme
+      }
+    }
+  private bindingContext: any;
+  public bind(bindingContext: any) {
+    this.bindingContext = bindingContext;
+    this.themeChanged(this.theme);
+    this.setViewportType();
+    window.addEventListener('resize', this);
+    this.positionChanged();
+    this.modalBreakpointChanged();
+    this.hostChanged();
+    this.overlayDismissChanged();
+    this.keyboardChanged();
+  }
+
+  public positionChanged() {
+    if (!this.position && this.defaultConfig.position) {
+      this.position = this.defaultConfig.position;
+    }
+  }
+
+  public modalBreakpointChanged() {
+    if (typeof this.modalBreakpoint !== 'number' && this.defaultConfig.modalBreakpoint) {
+      this.modalBreakpoint = this.defaultConfig.modalBreakpoint;
+    }
+  }
+
+  public hostChanged() {
+    if (this.host === false || this.host === 'body' || this.host instanceof HTMLElement) {
+      return;
+    }
+    if (this.defaultConfig.host !== undefined) {
+      this.host = this.defaultConfig.host;
+      return;
+    }
+    if (this.host === '') {
+      this.host = 'body';
+    }
+  }
+
+  public overlayDismissChanged() {
+    if (!this.overlayDismiss && this.defaultConfig.overlayDismiss) {
+      this.overlayDismiss = this.defaultConfig.overlayDismiss;
+    }
+  }
```

Comment:
we also have this option:
```suggestion
  public overlayDismissChanged(overlayDismiss) {
    if (!overlayDismiss && this.defaultConfig.overlayDismiss) {
      this.overlayDismiss = this.defaultConfig.overlayDismiss;
    }
  }
```
Basically all change handlers are called with signature: `(newValue, oldValue)`. It depends, just in case you are not aware of it.

Reply:
I know that changeHandler have the `(newValue, oldValue)` but for some reason I tend not to use it ?? Are there benefits of using the given value instead of the instance value ?

Reply:
generally parameter/local variables minifies better. and also it can avoid overhead of accessing a property. But I think it's not really necessary here. let's ignore this